clippy2021.md

```diff
@@ -0,0 +1,178 @@
+this is exactly what we need to do. Get together with the cargo team and write
+an RFC and implement such a configuration file somehow and somewhere.
+
+[#3164](https://github.com/rust-lang/rust-clippy/issues/3164)
+[cargo#5034](https://github.com/rust-lang/cargo/issues/5034)
+[IRLO](https://internals.rust-lang.org/t/proposal-cargo-lint-configuration/9135/8)
+
+
+#### FP rate
+
+In the worst case, new lints are only available in nightly for 2 weeks, before
+hitting beta and ultimately stable. This and the fact that fewer people use
+nightly Rust nowadays makes it more probable that a lint with many FPs hits
+stable. This leads to annoyed users, that will disable these new lints in the
+best case and to more annoyed users, that will stop using Clippy in the worst.
+We should think of a process to prevent this from happening. This will also be
+addressed later in this document.
+
+[#6429](https://github.com/rust-lang/rust-clippy/issues/6429)
+
+#### Clippy book
+
+We should consider writing a Clippy book, like other projects have. This will
+then be the central place where everything you need to know to use Clippy is
+collected in a structured way.
+
+[#6011](https://github.com/rust-lang/rust-clippy/issues/6011)
+
+### Reliability
+
+We should make sure to make Clippy more reliable and stable for our users
+
+#### FP rate
+
+In the worst case, new lints are only available in nightly for 2 weeks, before
+hitting beta and ultimately stable. This and the fact that fewer people use
+nightly Rust nowadays makes it more more probably that a lint with many FPs hits
+stable. This leads to annoyed users, that will disable these new lints in the
+best case and to more annoyed users, that will stop using Clippy in the worst.
+We should think of a process to prevent this from happening. This will also be
+addressed later in this issue.
+
+[#6429](https://github.com/rust-lang/rust-clippy/issues/6429)
+
+## Internal
+
+(The end of) 2020 has shown, that Clippy has to think about the available
+resources, especially regarding management and maintenance of the project. This
+section address issues affecting team members and contributors.
+
+### Management
+
+In 2020 we got over 1000 open issues with now between 25-35 open PRs. This is
+mostly because of the lack of resources in the team. According to the planned
+roadmap of the Rust project, we should improve our processes and look for more
+eyes and hands to help reviewing and triaging PRs and issues.
+
+#### Clear expectations for team members
+
+We will have to write this out according to the planned Rust 2021 roadmap
+anyway. This should not put more pressure on the team members, but rather help
+them and interested folks to know what our expectations are.
+
+#### More maintainer
+
+More people means less work for each individual. We should work out a process
+how onboarding of new folks can work and look out for people that are interested
+joining and taking up responsibility.
```

Comment:
Maybe we could try to create "working groups" to address the bigger issues stated in this document. One or more team members could provide mentorship on the issue to people that wants to get involved in the project. That way 1) we help folks getting acquainted with the project 2) some of them can turn into new maintainers 3) we fix a long-standing issue.

Related to this, when this document is finished I think we should try to get it announced in the Rust blog so that it gets more attention.

Reply:
> Maybe we could try to create "working groups" to address the bigger issues stated in this document

I think for every bigger project in this document a team member should be assigned and see to get it over the finish line, either with mentorship or implementing it themself. But let's focus on what we want to do here and address the "How" later.

> Related to this, when this document is finished I think we should try to get it announced in the Rust blog so that it gets more attention.

Yes, as @Mark-Simulacrum wrote, we can also include it in the Rust roadmap document.